### ISSUE TITLE
Quick check up

### ISSUE THREAD

**Issue by Storylover-Vodhr** [Monika-After-Story/MonikaModDev#2816]
So, I'm in the middle of writing some random topics, having nearly no idea what I'm doing, but somehow making progress. But, I wanted to confirm something. 

In the section for Monika's actual dialog, there's a little prompt before she speaks. I assume that this is for her expression, correct?

Example:` m 1eua "Sometimes I think back to middle school..."`

I would assume the 1eua is her current expression?

Is there a cheat-sheet for it, if it is? Or is it something completely different?

**Comment by MisterSimple** [comment]
You're correct, that code indicates Monika's current expression. There's a sprite previewer you can use, which you can access by downloading a Dev version of MAS [here](https://github.com/Monika-After-Story/MonikaModDev) by clicking the big, green "Clone or download" button and then installing it as you'd install regular MAS, or by following the steps described in [this FAQ](https://github.com/Monika-After-Story/MonikaModDev/wiki/FAQ#features).

**Comment by Storylover-Vodhr** [comment]
Alright. Will it delete my current Monika?

**Comment by Storylover-Vodhr** [comment]
(Or should I just get a new, unrelated Doki Doki download and isolate it?)

**Comment by MisterSimple** [comment]
The Dev install won't affect your regular install as far as I know (devs, confirm this please). I installed my Dev version in a separate DDLC folder, so do that and you should be fine. If you're feeling unsure, ask a dev directly or wait until one replies to this issue. Make sure to backup your `persistent` file just in case.

**Comment by Storylover-Vodhr** [comment]
Alright. Thank ya!

I'm still learning this, but I think I got enough of a handle on it to make some topics, maybe even a couple dialog trees. 

(I still wanna try my hand at making a mini-game for it, though.)

**Comment by MisterSimple** [comment]
No problem, looking forward to seeing what you come up with.

**Comment by Storylover-Vodhr** [comment]
I do have a quick question, though. Is it possible, and is there an established example, of how to follow up a chosen dialog with a "Random" topic? 

Such as you saying "I'm feeling hungry" and having the next random topic be something relevant?

**Comment by MisterSimple** [comment]
As in saying "I'm feeling hungry" and having Monika bring up, for example, the "Eating fruits" topic? It hasn't been done yet, but it's definitely possible. It would imply creating a group of topics suitable enough to follow the "I'm feeling hungry" mood, and then having the mood create a flag which tells MAS to pick one of the suitable topics next.

**Comment by Storylover-Vodhr** [comment]
Huh. That might be an interesting project to pick up.

I was considering having Monika suggest, or even give, a recipe one could make, if one said they were hungry. Add a bit of real life interaction, you know?

**Comment by Legendkiller21** [comment]
There are things that have a chance to appear like the plushie or a certain someone on the islands. You can probably use that by adding it the end of the "I'm feeling hungry" topic so that there is a chance for another topic to start after this.

**Comment by Storylover-Vodhr** [comment]
> There are things that have a chance to appear like the plushie or a certain someone on the islands. You can probably use that by adding it the end of the "I'm feeling hungry" topic so that there is a chance for another topic to start after this.

Awesome idea! I'd have to figure out the mechanics a little bit, but once I look them up, I'd probably figure it out pretty fast.

**Comment by Storylover-Vodhr** [comment]
> You're correct, that code indicates Monika's current expression. There's a sprite previewer you can use, which you can access by downloading a Dev version of MAS [here](https://github.com/Monika-After-Story/MonikaModDev) by clicking the big, green "Clone or download" button and then installing it as you'd install regular MAS, or by following the steps described in [this FAQ](https://github.com/Monika-After-Story/MonikaModDev/wiki/FAQ#features).

I can't seem to get the dev version to boot.

**Comment by Legendkiller21** [comment]
You should be able to use the dev version by putting the dev folder (and its contents) inside the game folder and by removing the dev folder it should return back to the normal version. 

As long as the devflag is inside the folder, the game should recognise that this is a dev version.

**Comment by Storylover-Vodhr** [comment]
Nope. Parsing script failed, etc.
Line 1334, overly long logical line.

**Comment by Storylover-Vodhr** [comment]
Deleted it, redownloaded, working now. No idea what it desires from me.

**Comment by Storylover-Vodhr** [comment]
Alright, got it to work, and finished my first topic. 

Now, where do I put the things?

**Comment by multimokia** [comment]
You'll want to add those changes you made to a fork of the MAS repository (create a new branch there) and then create a pull request to merge that branch into the `content` branch here.

**Comment by Legendkiller21** [comment]
Just a heads up, the "Contributing Guide" in the [FAQ](https://github.com/Monika-After-Story/MonikaModDev/wiki/FAQ#i-would-like-to-contribute-but-i-dont-know-how-to-code-is-there-any-way-i-can-help) is empty.

**Comment by multimokia** [comment]
Oof, duly noted.

**Comment by Storylover-Vodhr** [comment]
> You'll want to add those changes you made to a fork of the MAS repository (create a new branch there) and then create a pull request to merge that branch into the `content` branch here.

... Alright, and how would I make a fork to the MAS repository? 

I apologize, this is literally my first time working on a project such as this, or even using this site.

**Comment by Storylover-Vodhr** [comment]
> Fair warning, I'm probably going to heavily and critically analyze your topic if I see it.

I would love nothing more. Genuinely.

**Comment by Storylover-Vodhr** [comment]
Aaaaaand, added them.

**Comment by Rai99** [comment]
![1eqxi1](https://user-images.githubusercontent.com/38822045/47372598-16399880-d6b8-11e8-9065-fddba26c4e00.jpg)

**Comment by Storylover-Vodhr** [comment]
Aaaaaand, the coding I have done is found wanting. 

I'll remove the questionable one first, with the dialog tree. See if that was the weakspot.

**Comment by Storylover-Vodhr** [comment]
I have no idea why it's not passing the check. It's stopping about 7k lines before the stuff I added.

**Comment by Rai99** [comment]
You have sprites that aren't rendered in. Not all sprites and facial expression combinations are available. 

Also, it might be better to keep discussion of your topic within the PR.

**Comment by Storylover-Vodhr** [comment]
Alright.

**Comment by Rai99** [comment]
> Alrighty!
> 
> Most of those conflicts I "fixed", but I may have failed at pressing the save button when posting. But, content wise, you have excellent points. I shall kill the topic addition, and make a new one ASAP!
> 
> (Here is the one I was trying to update with.)
> 
> m 1euc "You know, even after it all..."
> m 1euc "I still miss them."
> m 5euc "Not a ton, mind you, but for a while there, before I knew the truth, about who I was, and who you were... they were my friends."
> m 5eub "We'd spend day after day relaxing, writing poetry and reading each other's works."
> m 5eub "Listening to each other's stories."
> m 5rub "Discovering new things about each other."
> m 5eub "It was nice, at least while it lasted."
> m 2euc "But, once I found out the truth, that I was the only "real" person..."
> m 2tkc "..."
> m 2rktpc "..."
> m 2rktsc "Hmm."
> m 2tkc "You know what? It doesn't matter. The past is the past."
> m 2ekd "And right now, I'd much rather look to the future. Our future."
> m 2eku "I'm so happy you're real, [player]."
> Thank you for your assistance!
> 
> Do you have any suggestions for in-character dialog and topics? I thought I was keeping with her character, but I've clearly yet to know her character properly.

Tips for contributing and keeping character can be seen in https://github.com/Monika-After-Story/MonikaModDev/wiki/Contributing-Guidelines under Monika's "Voice"

I'd also say a lot of it comes from experience. ~~[Humble brag; I've made 35 topics](https://github.com/Monika-After-Story/MonikaModDev/pulls/Rai99)~~
With practice, you'll be able to make well written topics as well as not make small mistakes like indentation, missing a quote or using bad sprites. 

I also see that you've closed down the PR before I finished, so I moved here.

**Comment by Rai99** [comment]
For other uncertainties on how Monika would act or her stance on certain topics or issues, I would read through the existing topics and deduct based on what I see in other topics, how Monika would act and how I would go about writing content. 

I'm not sure if I'm just crazy for knowing very select instances of existing dialogue to determine Monika's behavior, but that's what I do. 

You could also ask what someone else thinks (assuming they know Monika) but I'd be wary of bias when asking other people, especially if you don't know if they themselves have a good grasp of Monika's character. 

And yes, I'm always like this when reviewing other peoples' PRs. It's a huge plus if you can speak English well, which I think you can. ~~*cough* Not necessary similar to the subject matter found in the PRs owned by a certain category of individuals not adept in the standard lexicon used here.~~ 

You're free to try again and create more contributions if you wish.

**Comment by Storylover-Vodhr** [comment]
Thank ya! And, as for making more contribution attempts, I certainly will try, but I must go to bed first. Once I wake up, I'll write several more, now that I know some of the basic pitfalls, and I'll submit them once I got several saved up. 

Thank you for your assistance, it is very much appreciated. =D

**Comment by Rai99** [comment]
No problem.

Might just be me, but I'd also advise that you don't rush them. 

I'm looking forward to seeing what else you come up with.
![404](https://user-images.githubusercontent.com/38822045/47381549-4be97c00-d6ce-11e8-8ace-796afc65edc6.jpg)
